crates/ruff_server/src/server/api/requests/execute_command.rs

```diff
@@ -134,23 +145,66 @@ fn apply_edit(
     )
 }
 
-fn debug_information(session: &Session) -> String {
+fn debug_information(session: &Session, uri: Option<types::Url>) -> crate::Result<String> {
     let executable = std::env::current_exe()
         .map(|path| format!("{}", path.display()))
         .unwrap_or_else(|_| "<unavailable>".to_string());
-    format!(
-        "executable = {executable}
+
+    let mut buffer = String::new();
+
+    writeln!(
+        buffer,
+        "Global:
+executable = {executable}
 version = {version}
-encoding = {encoding:?}
-open_document_count = {doc_count}
-active_workspace_count = {workspace_count}
-configuration_files = {config_files:?}
-{client_capabilities}",
+position_encoding = {encoding:?}
+workspace_root_folders = {workspace_folders:#?}
+indexed_configuration_files = {config_files:#?}
+open_documents = {open_documents}
+client_capabilities = {client_capabilities:#?}
+",
         version = crate::version(),
         encoding = session.encoding(),
+        workspace_folders = session.workspace_root_folders().collect::<Vec<_>>(),
+        config_files = session.config_file_paths().collect::<Vec<_>>(),
+        open_documents = session.open_documents(),
         client_capabilities = session.resolved_client_capabilities(),
-        doc_count = session.num_documents(),
-        workspace_count = session.num_workspaces(),
-        config_files = session.list_config_files()
-    )
+    )?;
+
+    if let Some(uri) = uri {
+        let Some(snapshot) = session.take_snapshot(uri.clone()) else {
+            writeln!(buffer, "Unable to take a snapshot of the document at {uri}")?;
+            return Ok(buffer);
+        };
+
+        writeln!(
+            buffer,
+            "Document:
+uri = {uri}
+kind = {kind}
+version = {version}
+client_settings = {client_settings:#?}
+config_path = {config_path:?}
+{settings}
+            ",
+            uri = uri.clone(),
+            kind = match session.key_from_url(uri) {
+                DocumentKey::Notebook(_) => "Notebook",
+                DocumentKey::NotebookCell(_) => "NotebookCell",
+                DocumentKey::Text(_) => "Text",
+            },
+            version = snapshot.query().version(),
+            client_settings = snapshot.client_settings(),
+            config_path = snapshot.query().settings().path(),
+            settings = snapshot.query().settings(),
```

Comment:
Nit: You may want to store `query` in a temporary variable to shorten some of those calls.